### ISSUE TITLE
Zed X11 CSD behavior should follow of most other programs

### ISSUE THREAD

**Issue by FreeAoi** [zed-industries/zed#17623]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

With current Zed x11 CSD implementation it check [2 things](https://github.com/zed-industries/zed/blob/f71cb14d7a3c876b1a5a69949a5a198ed3bfbd5a/crates/gpui/src/platform/linux/x11/client.rs#L301) in order to know if CSD can be used:

- [if there is a compositor](https://github.com/zed-industries/zed/blob/f71cb14d7a3c876b1a5a69949a5a198ed3bfbd5a/crates/gpui/src/platform/linux/x11/client.rs#L1570)
- [if _GTK_FRAME_EXTENDS is supported](https://github.com/zed-industries/zed/blob/f71cb14d7a3c876b1a5a69949a5a198ed3bfbd5a/crates/gpui/src/platform/linux/x11/client.rs#L1642)

CSD can be divided in drawing the titlebar and drawing the shadows. For zed be avaliable to drawing the titlebar it should check for another property called `_NET_FRAME_EXTENTS` instead of `_GTK_FRAME_EXTENDS` since this should be only check when trying to set shadows, for example chromium only check for `_GTK_FRAME_EXTENDS` to set shadows [ref](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/x11/x11_window.cc;l=1090-1108?q=_GTK_FRAME_EXTENTS&ss=chromium) [ref2](https://source.chromium.org/chromium/chromium/src/+/main:ui/ozone/platform/x11/x11_window.cc;l=1413-1434?q=_GTK_FRAME_EXTENTS&ss=chromium) also not only chromium follows this approach.

Check example above Chrome with custom title bar
![image](https://github.com/user-attachments/assets/ebad0e67-d741-461c-96cb-f70a74a3d7a8)

I'm using openbox and _GTK_FRAME_EXTENTS is not supported but it should be avaliable to draw the custom titlebar check chrome with the custom titlebar, the actual behavior is zed using the server decoration
![image](https://github.com/user-attachments/assets/5b1011a8-d7bc-4630-b75f-41dd0ba18728)

**In summary Zed should disable just shadows when _GTK_FRAME_EXTENTS is not supported but still draw the titlebar**

### Environment

Zed: v0.152.1 (Zed Preview)
OS: Linux X11 archcraft unknown
Memory: 7.5 GiB
Architecture: x86_64
GPU: Intel(R) Xe Graphics (TGL GT2) || Intel open-source Mesa driver || Mesa 24.2.2-arch1.1

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>

</pre></details>
